### ISSUE TITLE
WSI Registration Support

### ISSUE THREAD

**Issue by eshasadia** [uncbiag/uniGradICON#23]
Hi, Is there anyway to try this model on whole slide images? I am currently working on restrained and consecutive wsi slides  and would like to see the result of this algorithm

**Comment by lintian-a** [comment]
Hi @eshasadia ,

Thanks for your interest on trying uniGradICON on your data! 

I haven't worked with WSI slides before and haven't tested uniGradICON on WSI slides. Thus, I am not sure how it would perform. There are currently three ways to run uniGradICON/multiGradICON:
- [3D Slicer extension](https://github.com/uncbiag/SlicerUniGradICON)
- [Command line](https://github.com/uncbiag/uniGradICON?tab=readme-ov-file#easy-to-use-and-install)
- [Colab notebook](https://colab.research.google.com/drive/1JuFL113WN3FHCoXG-4fiBTWIyYpwGyGy?usp=sharing)

What file format are the images stored in? And are they 3D images? 

If none of the above interfaces work in your case, what customizations are needed for the WSI slides?

**Comment by marcniethammer** [comment]
Also, uniGradICON is trained for 3D whereas I assume your WSI are in 2D. GradICON can be trained for 2D as well (but we don't have a general model for it).

**Comment by eshasadia** [comment]
yes, I have multi stained 2D WSI files, is there anyway to try this model on them.

**Comment by marcniethammer** [comment]
This model is currently focused on 3D. For 2D a starting point could be #17 
But ultimately it would be useful to train a dedicated 2D model.

**Comment by HastingsGreer** [comment]
GradICON can train on images up to 4096 x 4086 on an A100. Here is a notebook demonstrating this on a toy dataset:

https://colab.research.google.com/drive/1BoWHAYhMwzSfj5BKxcEOSbcYUjm9ehof?usp=sharing

I would be very curious to know how well this training works if you swap in your data for the toy dataset used in the notebook, and happy to support if you run in to issues (or success!)
